src/Microsoft.Diagnostics.EventFlow.Core/Implementations/DiagnosticPipeline.cs

```diff
@@ -229,16 +290,19 @@ public EventData[] FilterAndDecorate(EventData[] eventsToFilter)
                     return null;
                 }
 
-                // CONSIDER: using a Queue<T> with external lock and be able to reuse the queue
-                // (ConcurrentQueue has no Clear() method while ordinary Queue does).
-                ConcurrentQueue<EventData> eventsToKeep = new ConcurrentQueue<EventData>();
+                ConcurrentStack<EventData> eventsToKeep;
```

Comment:
We do not guarantee the order of the events in the pipeline will be maintained, so from correctness perspective the change is fine. Actually, there is no guarantee that events will be submitted to the pipeline in any particular order. Nor do we guarantee that the batches of events are sent out in order by outputs.

Each event has a timestamp so we know what happened when.